src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/datasource/OffenderDetailsDataSource.kt

```diff
@@ -42,12 +43,20 @@ class ConfiguredOffenderDetailsDataSource(
 }
 
 @Component
-class CommunityApiOffenderDetailsDataSource : OffenderDetailsDataSource {
+class CommunityApiOffenderDetailsDataSource(
+  private val communityApiClient: CommunityApiClient,
+) : OffenderDetailsDataSource {
   override val name: OffenderDetailsDataSourceName
     get() = OffenderDetailsDataSourceName.COMMUNITY_API
 
   override fun getOffenderDetailSummary(crn: String): ClientResult<OffenderDetailSummary> {
-    throw NotImplementedError("Getting details for individual offenders from the Community API is not currently supported")
+    var offenderResponse = communityApiClient.getOffenderDetailSummaryWithWait(crn)
+
+    if (offenderResponse is ClientResult.Failure.PreemptiveCacheTimeout) {
```

Comment:
I'm hoping we'll be able to make the preemptive caching go away soon, but it makes sense to keep it in place for the Community API calls in the meantime